tensorflow/lite/delegates/webnn/webnn_delegate.cc

```diff
@@ -399,6 +400,75 @@ class Subgraph {
     }
   }
 
+  // Calculate WebNN padding values specific for kTfLitePaddingSame.
+  static TfLiteStatus CalculateWebnnPaddings(
```

Comment:
Oh, so I guess TF *didn't* already have a helpful existing function to compute this :(.

Reply:
They do have the helper function, it's [`ComputePaddingHeightWidth`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/padding.h#L62).

`CalculateWebnnPaddings`: this function is used for handling the difference between `TransposeConv` and `Conv`, as `TransposeConv` uses `output_height` and `output_width` to calculate expected input height and input width.

Reply:
I see. I just saw this sizeable function `CalculateWebnnPaddings` and thought you had to write a bunch of new logic yourself, but now I see that *most* of it here is actually error checking.